cirq-google/cirq_google/serialization/arg_func_langs.py

```diff
@@ -133,15 +134,22 @@ def arg_to_proto(
         msg.arg_value.bool_value = bool(value)
     elif isinstance(value, FLOAT_TYPES):
         msg.arg_value.float_value = float(value)
+    elif isinstance(value, complex):
+        msg.arg_value.complex_value.real_value = value.real
+        msg.arg_value.complex_value.imaginary_value = value.imag
     elif isinstance(value, bytes):
         msg.arg_value.bytes_value = value
     elif isinstance(value, str):
         msg.arg_value.string_value = value
     elif isinstance(value, (list, tuple, np.ndarray)):
         if len(value):
```

Comment:
On a round trip this converts (i) uniform-type lists or tuples to lists and (ii) non-uniform lists and tuples to tuples.  It also discards empty lists, tuples and arrays, which return as None from round trip.

We can move the array conversion out of the `if len(value)` block as `_ndarray_to_proto` and `_ndarray_from_proto` seem to work with empty arrays.

As for tuples and lists, we could add a `sequence_type` enum to the `ArgValue` message which could be `(UNSPECIFIED, LIST, TUPLE)` and would let us reconstruct either `list` or `tuple` from a round trip.  It would also allow us to express either an empty list or an empty tuple, in such case `ArgValue` would be empty except of a (new) `sequence_type`.

Reply:
Good idea!  I went ahead and did that, and also added set support while in there.